data/src/server.rs

```diff
@@ -54,6 +55,13 @@ impl Map {
         self.0.iter().map(Entry::from)
     }
 
+    pub fn update_proxy(&mut self, proxy: &Option<ProxyConfig>) -> Result<(), Error> {
+        for (_, config) in self.0.iter_mut() {
+            config.proxy = proxy.clone();
+        }
+        Ok(())
+    }
```

Comment:
Shouldn't a server level proxy config take precedence over a root proxy config? One can simply delete the server proxy to have the root one take effect.

I'd say we only set this if it's None

Reply:
the current implementation just supports a global proxy configuration. having a server specific proxy config would also be a good suggestion that we can implement and use that with a higher priority.

Reply:
I'm confused, we have ProxyConfig specified in the server config, so users can set the proxy in toml for each server. Is that not intended?

Reply:
I've fixed in efd7b54 to remove this "injection" and reference the proxy config from a single source of truth